osutil/mockable.go

```diff
@@ -20,17 +20,23 @@
 package osutil
 
 import (
-	"os"
-	"os/user"
-	"syscall"
+	"bytes"
+	"io"
 )
 
-// MockMountInfo is meant for tests to mock the content of /proc/self/mountinfo.
+type mountInfoMockFile struct{ *bytes.Buffer }
+
+func (f *mountInfoMockFile) Close() error {
+	return nil
+}
+
 func MockMountInfo(content string) (restore func()) {
-	old := mockedMountInfo
-	mockedMountInfo = &content
+	old := openMountInfoFile
+	openMountInfoFile = func() (io.ReadCloser, error) {
+		return &mountInfoMockFile{bytes.NewBufferString(content)}, nil
```

Comment:
```suggestion
		return ioutil.NopCloser(bytes.NewBufferString(content)), nil
```
and you can drop mockInfoMockFile

Reply:
Wow, this is great. I couldn't even imagine that such functionality is already builtin. Thank you so much!

Are there some good sources to read/learn things like this one?

Reply:
I'm not aware of any sources, but io interfaces are very common in go, and some useful helpers built around those are in io/ioutil and then there are other abstractions built on top like for example: https://pkg.go.dev/net/http?utm_source=godoc#MaxBytesReader

Reply:
For those that might look for the answer in the future regarding some additional source of information about effective features similar to `ioutil.NopCloser` and other guidelines for effective Go. The following source was suggested as a good reference for this kind of stuff: https://go.dev/doc/effective_go